### PR TITLE
BUG: Series constructor raising with NA and bools and Int dtype

### LINKED ISSUE #42137: BUG/API: constructing nullable integer and floating arrays from boolean data
From https://github.com/pandas-dev/pandas/pull/41970#discussion_r654770580, right now constructing nullable integer (and floating) arrays from boolean data has the following behavior:

```
In [2]: pd.Series([True, False, True, True], dtype='Int64')
Out[2]:
0    1
1    0
2    1
3    1
dtype: Int64

In [3]: pd.Series([True, False, True, pd.NA], dtype='Int64')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
...
TypeError: object cannot be converted to an IntegerDtype
```

This seems inconsistent, if so, which behavior is correct?

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #42137 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Converting here makes more sense than raising imo

### CODE REVIEW

File: pandas/tests/series/test_constructors.py [hunk 1]
```diff
@@ -1987,6 +1987,19 @@ def test_constructor_mismatched_null_nullable_dtype(
...
+        expected = Series([1, 0, 1, pd.NA], dtype="Int64")
+        tm.assert_series_equal(result, expected)
+
+        result = Series([True, False, True], dtype="Int64")
+        expected = Series([1, 0, 1], dtype="Int64")
+        tm.assert_series_equal(result, expected)
+
+        with pytest.raises(ValueError, match="invalid literal"):
+            Series(["True", "False", "True", pd.NA], dtype="Int64")
+
```
Comment (gfyoung) [inline pandas/tests/series/test_constructors.py:2002]: I'd split these into separate tests

Comment (phofl) [reply pandas/tests/series/test_constructors.py:2004]: Moved into separate test

**Comment by gfyoung** [comment]
LGTM, not sure what's going on with GH Actions

File: doc/source/whatsnew/v2.0.0.rst [hunk 2]
```diff
@@ -812,6 +812,7 @@ Numeric
 Conversion
 ^^^^^^^^^^
 - Bug in constructing :class:`Series` with ``int64`` dtype from a string list raising instead of casting (:issue:`44923`)
+- Bug in constructing :class:´Series` with masked dtype and boolean values with ``NA`` raising (:issue:`42137`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:815]:
```suggestion
- Bug in constructing :class:`Series` with masked dtype and boolean values with ``NA`` raising (:issue:`42137`)
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**